src/shogun/machine/FeatureDispatchCRTP.h

```diff
@@ -0,0 +1,120 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Shubham Shukla
+ */
+
+#ifndef _FEATUREDISPATCHCRTP_H__
+#define _FEATUREDISPATCHCRTP_H__
+
+#include <shogun/lib/config.h>
+
+#include <shogun/features/DenseFeatures.h>
+#include <shogun/lib/common.h>
+#include <shogun/machine/LinearMachine.h>
+
+namespace shogun
+{
+	class CFeatures;
+	class CLabels;
+#define IGNORE_IN_CLASSLIST
+
+	template <class P, class T>
+	IGNORE_IN_CLASSLIST class CDenseFeaturesDispatch : public T
+	{
+	public:
+		/** Default constructor */
+		CDenseFeaturesDispatch() : T()
+		{
+		}
+
+		virtual ~CDenseFeaturesDispatch()
+		{
+		}
+
+	protected:
+		virtual bool train_dense(CFeatures* data)
+		{
+			REQUIRE(
+			    data->get_num_vectors() == this->m_labels->get_num_labels(),
```

Comment:
the check should happen inside `CMachine`, otherwise we have to repeat it

Reply:
yes since we are enforcing data should be passed during train